CMakeLists.txt

```diff
@@ -1,11 +1,144 @@
+        INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+        COMPONENT creaderwriterqueue_RunTime
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        COMPONENT readerwriterqueue_RunTIme
+        NAMELINK_COMPONENT readerwriterqueue_Development
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        COMPONENT readerwriterqueue_Development
+)
+
+set(readerwriterqueue_INSTALL_CMAKEDIR
+        ${CMAKE_INSTALL_DATADIR}/readerwriterqueue
+        CACHE STRING "Path to readerwriterqueue cmake files"
+)
+
+install(EXPORT readerwriterqueue_Targets
+        DESTINATION ${readerwriterqueue_INSTALL_CMAKEDIR}
+        NAMESPACE moodycamel::
+        FILE readerwriterqueueTargets.cmake
+        COMPONENT readerwriterqueue_Development
+)
+
+include(CMakePackageConfigHelpers)
+configure_package_config_file(${CMAKE_CURRENT_SOURCE_DIR}/cmake/config.cmake.in
+        "${CMAKE_CURRENT_BINARY_DIR}/readerwriterqueueConfig.cmake"
+        INSTALL_DESTINATION ${readerwriterqueue_INSTALL_CMAKEDIR}
+)
+
+
+get_target_property(readerwriterqueue_VERSION readerwriterqueue VERSION)
+
+write_basic_package_version_file(readerwriterqueueConfigVersion.cmake VERSION ${readerwriterqueue_VERSION} COMPATIBILITY SameMajorVersion)
+
+install(FILES
+        ${CMAKE_CURRENT_BINARY_DIR}/readerwriterqueueConfig.cmake
+        ${CMAKE_CURRENT_BINARY_DIR}/readerwriterqueueConfigVersion.cmake
+        DESTINATION ${readerwriterqueue_INSTALL_CMAKEDIR}
+)
+
+
+install(FILES
+        atomicops.h
+        readerwriterqueue.h
+        readerwritercircularbuffer.h
+        LICENSE.md
         DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME})
+
+
+option(MOODYCAMEL_READERWRITERQUEUE_ENABLE_TESTS "Enables readerwriterqueue tests" OFF)
+
+if(${MOODYCAMEL_READERWRITERQUEUE_ENABLE_TESTS})
+
+    find_package(Threads REQUIRED)
+#    add_library(simple)
+    add_executable(unittests)
+    target_sources(unittests PRIVATE
+            tests/unittests/minitest.h
+            tests/unittests/unittests.cpp
+            tests/common/simplethread.h
+            tests/common/simplethread.cpp
+    )
+    target_include_directories(unittests PRIVATE
+            tests/unittests/)
+#    static.
+    set_property(TARGET unittests PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
+    # handles pthread and such.
+    target_link_libraries(unittests PRIVATE Threads::Threads)
+#    target_link_libraries(unittests PRIVATE readerwriterqueue)
+    target_compile_options(unittests
+            PRIVATE
+            $<$<CXX_COMPILER_ID:Clang>: -Wsign-conversion -Wpedantic -Wall -DNDEBUG -O3 -g>
+            $<$<CXX_COMPILER_ID:GNU>:  -Wsign-conversion -Wpedantic -Wall -DNDEBUG -O3 -g>
+#            /W4 is similar to -Wall, next are similar to -sign-conversion, /permissive- is similar to -Wpedantic.
+            $<$<CXX_COMPILER_ID:MSVC>: /W4 /w14287 /w14826 /permissive- /02 /DEBUG>
```

Comment:
Why are the unit tests compiled in debug mode for MSVC only?

Reply:
Think I was mislead by `/DEBUG` and the lack of `/DNDEBUG`. This makes more sense. Can you add `/DNDEBUG` for MSVC too?

Reply:
I realized that I missed it and put it in the latest changes.  normally it's not necessary to add that definition as it is automatically added in appropriate CMake configs, however I added anyway, though this time under the correct usage target_compile_definitions